### PR TITLE
feat(application): enrich helm install and uprade options

### PR DESCRIPTION
**Pull request by rootdeep**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
kind feature

**What this PR does / why we need it**:

Helm install and helm upgrade owns public options with different values. Extracting  paramters into app.spec.Chart.InstallPara and app.spec.Chart.UpradePara to avoid interaction. 
A demo result as bellow:

image

### CODE REVIEW

File: api/application/v1/types.go [hunk 1]
```diff
@@ -77,15 +78,39 @@ type Chart struct {
...
+	HelmPublicPara `json:",inline" protobuf:"bytes,1,opt,name=helmPublicPara"`
+}
+
+//public parameters used in helm install and helm upgrade command
+type HelmPublicPara struct {
+	//Client timeout when installiing or upgrading helm release, override default clientTimeOut
+	Timeout time.Duration `json:"clientTimeout" protobuf:"bytes,1,opt,name=clientTimeout"`
 	// CreateNamespace create namespace when install helm release
-	CreateNamespace bool `json:"createNamespace" protobuf:"bytes,9,opt,name=createNamespace"`
+	CreateNamespace bool `json:"createNamespace" protobuf:"bytes,2,opt,name=createNamespace"`
```
Comment (wl-chen) [inline api/application/v1/types.go:105]: 这个protobuf协议的顺序改变，会不会影响现有的application数据？

Comment (huxiaoliang) [reply api/application/v1/types.go:105]: I concern about it this as well

**Review by DanielXLee — APPROVED** [review]
/lgtm

**Review by GaoXiaodong — APPROVED** [review]

File: pkg/application/controller/app/action/install.go [hunk 2]
```diff
@@ -81,16 +81,23 @@ func Install(ctx context.Context,
 		return nil, err
 	}
 
+	var clientTimeout = defaultTimeout
```
Comment (leoryu) [inline pkg/application/controller/app/action/install.go:84]: is this nessary?

File: pkg/application/controller/app/action/install.go [hunk 3]
```diff
@@ -81,16 +81,23 @@ func Install(ctx context.Context,
...
 	_, err = client.Install(ctx, &helmaction.InstallOptions{
 		Namespace:        newApp.Spec.TargetNamespace,
 		ReleaseName:      newApp.Spec.Name,
-		Atomic:           newApp.Spec.Chart.Atomic,
-		CreateNamespace:  newApp.Spec.Chart.CreateNamespace,
+		CreateNamespace:  newApp.Spec.Chart.InstallPara.CreateNamespace,
 		DependencyUpdate: true,
 		Values:           values,
-		Timeout:          clientTimeOut,
+		Timeout:          clientTimeout,
```
Comment (leoryu) [inline pkg/application/controller/app/action/install.go:96]: why not just use defaultTimeout?

File: pkg/application/controller/app/action/install.go (hunk 2 above)
Comment (rootdeep) [reply pkg/application/controller/app/action/install.go:84]: why not? Usually we don't set timeout in installPara or upgradePara, and we use the defaultTimeout.

Comment (leoryu) [reply pkg/application/controller/app/action/install.go:84]: ok

**Review by leoryu — APPROVED** [review]

**Pull request merged**